app/models/user.rb

```diff
@@ -1,6 +1,23 @@
+# frozen_string_literal: true
+
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+    :recoverable, :rememberable, :validatable
+
+  has_many :donor_contacts, dependent: :destroy
+  has_many :donations, dependent: :nullify
+
+  scope :donors, -> { where(is_donor: true) }
+  scope :admins, -> { where(is_admin: true) }
```

Comment:
These let you get only a selection of `User`s via `User.donors` or `User.admins` :)